docs/en/operations/settings/query-complexity.md

```diff
@@ -8,465 +8,110 @@ title: 'Restrictions on Query Complexity'
-        <max_sessions_for_user>0</max_sessions_for_user>
-    </unlimited_sessions_profile>
-</profiles>
-<users>
-     <!-- User Alice can connect to a ClickHouse server no more than once at a time. -->
-    <Alice>
-        <profile>single_session_user</profile>
-    </Alice>
-    <!-- User Bob can use 2 simultaneous sessions. -->
-    <Bob>
-        <profile>two_sessions_profile</profile>
-    </Bob>
-    <!-- User Charles can use arbitrarily many of simultaneous sessions. -->
-    <Charles>
-       <profile>unlimited_sessions_profile</profile>
-    </Charles>
-</users>
-```
-
-Default value: 0 (Infinite count of simultaneous sessions).
-
-## max_partitions_to_read {#max-partitions-to-read}
-
-Limits the maximum number of partitions that can be accessed in one query.
-
-The setting value specified when the table is created can be overridden via query-level setting.
-
-Possible values:
-
-- Any positive integer.
-
-Default value: -1 (unlimited).
-
-You can also specify a MergeTree setting [max_partitions_to_read](merge-tree-settings#max-partitions-to-read) in tables' setting.
+## Overview {#overview}
+
+As part of the [settings](/operations/settings/overview), ClickHouse offers
+the ability to place restrictions on query complexity. This helps protect against
+potentially resource-intensive queries, ensuring safer and more predictable 
+execution, particularly when using the user interface.
+
+Almost all the restrictions only apply to `SELECT` queries, and for distributed 
+query processing, restrictions are applied on each server separately.
+
+ClickHouse generally checks the restrictions only after data parts have been 
+fully processed, rather than checking the restrictions for each row. This can
+result in a situation where restrictions are violated while the part is being
+processed.
+
+## `overflow_mode` settings {#overflow_mode_setting}
+
+Most restrictions also have an `overflow_mode` setting, which defines what happens
+when the limit is exceeded, and can take one of two values:
+- `throw`: throw an exception (default).
+- `break`: stop executing the query and return the partial result, as if the 
+           source data ran out.
+
+## `group_by_overflow_mode` settings {#group_by_overflow_mode_settings}
+
+The `group_by_overflow_mode` setting also has
+the value `any`:
+- `any` : continue aggregation for the keys that got into the set, but do not 
+          add new keys to the set.
+
+## List of settings {#relevant-settings}
+
+The following settings are used for applying restrictions on query complexity.
+
+:::note
+Restrictions on the "maximum amount of something" can take a value of `0`,
+which means that it is "unrestricted".
+:::
+
+| Setting                                                                                                                | Short description                                                                                                                                                |
```

Comment:
Added this table, but unsure whether to keep or remove. On the one hand it provides a good overview, but on the other hand it will get out of date when people add new settings to Settings.cpp and forget to update it.

We can probably generate it from the source code at some point.